lib/pbench/server/api/resources/upload_api.py

```diff
@@ -124,20 +121,32 @@ def put(self, controller):
         md5_full_path = Path(path, f"{filename}.md5")
         bytes_received = 0
 
-        # TODO: Need real user from PUT!
-
         # Create a tracking dataset object; it'll begin in UPLOADING state
         try:
             dataset = Dataset(
                 owner=username, controller=controller, path=tar_full_path, md5=md5sum
             )
             dataset.add()
+        except DatasetDuplicate:
+            self.logger.info("Dataset already exists {}", filename)
+            abort(
+                200, message="Dataset already exists",
+            )
         except Exception:
             self.logger.exception("unable to create dataset for {}", filename)
             abort(
                 HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR",
             )
 
+        if tar_full_path.is_file() and md5_full_path.is_file():
+            self.logger.error(
+                "Dataset or corresponding md5 file already present on the disc, {}",
```

Comment:
What you've coded is "tarball __and__ MD5 file" not "or". I wonder why?

Reply:
Ah yeah it should be `or`